bundles/org.openhab.binding.vesync/src/main/resources/OH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,341 @@
+				<description>The name allocated to the device by the app. (Must be unique if used)</description>
+			</parameter>
+		</config-description>
+
+	</thing-type>
+
+	<channel-type id="deviceEnabledType">
+		<item-type>Switch</item-type>
+		<label>Switched On</label>
+		<description>Indicator if the device is switched on</description>
+		<state readOnly="false"/>
+	</channel-type>
+
+	<channel-type id="deviceChildLockEnabledType">
+		<item-type>Switch</item-type>
+		<label>Display Lock</label>
+		<description>Indicator if the devices child lock is enabled (Display Lock)</description>
+		<state readOnly="false"/>
+	</channel-type>
+
+	<channel-type id="deviceDisplayEnabledType">
+		<item-type>Switch</item-type>
+		<label>Display</label>
+		<description>Indicator if the devices display is enabled</description>
+		<state readOnly="false"/>
+	</channel-type>
+
+	<channel-type id="deviceFilterLifePercentageType">
+		<item-type>Number:Dimensionless</item-type>
+		<label>Filter Life Remaining</label>
+		<description>Indicator of the remaining filter life</description>
+		<state readOnly="true" pattern="%.0f %%"/>
+	</channel-type>
+
+	<channel-type id="airPurifierModeType">
+		<item-type>String</item-type>
+		<label>Operation Mode</label>
+		<description>The operating mode the air purifier is currently set to</description>
+		<state readOnly="false">
+			<options>
+				<option value="auto">Auto</option>
+				<option value="manual">Manual Fan Control</option>
+				<option value="sleep">Sleeping Auto</option>
+			</options>
+		</state>
+	</channel-type>
+
+	<channel-type id="deviceAFNightLight">
+		<item-type>String</item-type>
+		<label>Night Light</label>
+		<description>The operating mode of the night light functionality</description>
+		<state readOnly="false">
+			<options>
+				<option value="on">On</option>
+				<option value="dim">Dimmed</option>
+				<option value="off">Off</option>
+			</options>
+		</state>
+	</channel-type>
+
+
+	<channel-type id="airPurifierFanLevelType">
+		<item-type>Number:Dimensionless</item-type>
+		<label>Fan Speed</label>
+		<description>Indicator of the current fan speed</description>
+		<state readOnly="true" pattern="%.0f"/>
+	</channel-type>
+
+	<channel-type id="deviceErrorCodeType">
+		<item-type>Number:Dimensionless</item-type>
+		<label>Device Error Code</label>
+		<description>Indicator of the current error code of the device</description>
+		<state readOnly="true" pattern="%.0f"/>
+	</channel-type>
+
+	<channel-type id="deviceAirQualityBasicType">
+		<item-type>Number:Dimensionless</item-type>
+		<label>Air Quality</label>
+		<description>System representation of air quality</description>
+		<state readOnly="true" pattern="%.0f"/>
+	</channel-type>
+
+	<channel-type id="airQualityPPM25">
+		<item-type>Number:Dimensionless</item-type>
+		<label>Air Quality PPM2.5</label>
+		<description>Indicator of current air quality</description>
+		<state readOnly="true" pattern="%.0f PPM 2.5"/>
```

Comment:
@fwolter Could do with further input here - sorry - and thanks again for all the time looking through this!!! I was just thinking about this one, as it jogged memories from when this was done last year. So the VeSync app and physical devices report the air quality as a unit from 0 - 999 in PM 2.5 which I assume is particles per 2.5 million. Really for the units to be PPM, the reported value would have to be manipulated to the corresponding scale. With my user hat on, its much simpler to write rules based on the units / app's display, also as a user I would think the openHAB interface is incorrect if it was giving a different displayed number to what the app and unit reports as coming from the device? I'm not sure whether it should be adjusted or not - but I know from using these unit's myself it would be much harder to figure out the rule's /  conditions if the numbers were scaled to ppm? I'd personally probably opt for not manipulating the reported numbers to a scaled PPM, just to save people thinking the openHAB interface is wrong, and by using the devices unit's it simplifies the creation of rules, as I know what a PM2.5 reading of 15 is but wouldn't know the corresponding PPM value easily. If the numbers are not scaled then a ppm unit isn't correct as its got the 2.5 scaler attribute as part of it?

Reply:
PPM2.5 is a common value to measure fine particles. It means 2.5µm-sized particles per million. See https://www.health.ny.gov/environmental/indoors/air/pmq_a.htm

So, you can take the VeSync value as is and simply use ppm as the unit.

Reply:
Units corrected now :)